lib/grape-swagger/endpoint.rb

```diff
@@ -287,17 +287,13 @@ def expose_params_from_model(model)
       model_name
     end
 
-    def model_name(name)
-      if name.respond_to?(:entity_name)
-        name.entity_name
-      elsif name.to_s.end_with?('Entity', 'Entities')
-        length = 0
-        name.to_s.split('::')[0..-2].reverse.take_while do |x|
-          length += x.length
-          length < 42
-        end.reverse.join
+    def model_name(entity)
+      if entity.respond_to?(:entity_name)
+        entity.entity_name
+      elsif entity.to_s.end_with?('::Entity', '::Entities')
+        entity.to_s.split('::')[-2]
       else
-        name.name.demodulize.camelize
+        entity.name.demodulize.camelize
```

Comment:
Take the example again of animals:

`AnimalFarm::Internal::CowEntity` and an `AnimalFarm::Public::CowEntity`, the internal is only used by internal apps and exposes all attributes, the public one only the important attributes. Using this they both end up being `CowEntity` and depending on the load order the swagger output will show the public or internal